testsuite/standalone/src/test/java/org/wildfly/extras/creaper/commands/elytron/tls/AddFilteringKeyStoreOnlineTest.java

```diff
@@ -0,0 +1,171 @@
+                KEY_STORE_NAME, FILTER_ALIAS)
+                .build();
+        assertFalse("The filtering key store should not exist", ops.exists(FILTERING_KEY_STORE_ADDRESS));
+        client.apply(addFilteringKeyStore);
+        assertTrue("Filtering key store should be created", ops.exists(FILTERING_KEY_STORE_ADDRESS));
+    }
+
+    @Test
+    public void addTwoSimpleFilteringKeyStores() throws Exception {
+        AddFilteringKeyStore addFilteringKeyStore = new AddFilteringKeyStore.Builder(FILTERING_KEY_STORE_NAME,
+                KEY_STORE_NAME, FILTER_ALIAS)
+                .build();
+        AddFilteringKeyStore addFilteringKeyStore2 = new AddFilteringKeyStore.Builder(FILTERING_KEY_STORE_NAME2,
+                KEY_STORE_NAME, FILTER_ALIAS)
+                .build();
+
+        assertFalse("The filtering key store should not exist", ops.exists(FILTERING_KEY_STORE_ADDRESS));
+        assertFalse("The filtering key store should not exist", ops.exists(FILTERING_KEY_STORE_ADDRESS2));
+
+        client.apply(addFilteringKeyStore);
+        client.apply(addFilteringKeyStore2);
+
+        assertTrue("Filtering key store should be created", ops.exists(FILTERING_KEY_STORE_ADDRESS));
+        assertTrue("Filtering key store should be created", ops.exists(FILTERING_KEY_STORE_ADDRESS2));
+    }
+
+    @Test(expected = CommandFailedException.class)
+    public void addDuplicatefilteringKeyStoreNotAllowed() throws Exception {
+        AddFilteringKeyStore addFilteringKeyStore = new AddFilteringKeyStore.Builder(FILTERING_KEY_STORE_NAME,
+                KEY_STORE_NAME, FILTER_ALIAS)
+                .build();
+        AddFilteringKeyStore addFilteringKeyStore2 = new AddFilteringKeyStore.Builder(FILTERING_KEY_STORE_NAME,
+                KEY_STORE_NAME, FILTER_ALIAS)
+                .build();
+
+        client.apply(addFilteringKeyStore);
+        assertTrue("The filtering key store should be created", ops.exists(FILTERING_KEY_STORE_ADDRESS));
+
+        client.apply(addFilteringKeyStore2);
+        fail("Filtering key store is already configured, exception should be thrown");
+    }
+
+    @Test
+    public void addDuplicateFilteringKeyStoreAllowed() throws Exception {
+        AddFilteringKeyStore addFilteringKeyStore = new AddFilteringKeyStore.Builder(FILTERING_KEY_STORE_NAME,
+                KEY_STORE_NAME, FILTER_ALIAS)
+                .build();
+        AddFilteringKeyStore addFilteringKeyStore2 = new AddFilteringKeyStore.Builder(FILTERING_KEY_STORE_NAME,
+                KEY_STORE_NAME, FILTER_ALIAS)
+                .replaceExisting()
+                .build();
+
+        client.apply(addFilteringKeyStore);
+        assertTrue("The filtering key store should be created", ops.exists(FILTERING_KEY_STORE_ADDRESS));
+
+        client.apply(addFilteringKeyStore2);
+        assertTrue("The filtering key store should be created", ops.exists(FILTERING_KEY_STORE_ADDRESS));
```

Comment:
Use also checkAttribute method to ensure that some value was really changed after replacing.